### PR TITLE
Fix type checking of compound assignments

### PR DESCRIPTION
**Pull request by clinssen**
This fixes type checking of ``*=`` and ``/=`` compound assignments. The criteria for type correctness were incorrectly formulated.

### CODE REVIEW

**Comment by clinssen** [comment]
#654 should be merged or closed before this PR.

**Comment by clinssen** [comment]
@pnbabu: the latest commits should address all your comments, could you re-review? Cheers!

File: tests/unit_system_test.py [hunk 1]
```diff
@@ -111,6 +111,13 @@ def test_expression_after_magnitude_conversion_in_compound_assignment(self):
         printed_rhs_expression = print_rhs_of_first_assignment_in_update_block(model)
         self.assertEqual(printed_rhs_expression, '0.001 * (1200*mV)')
 
+    def test_expression_after_magnitude_conversion_in_compound_mult_div(self):
+        model = ModelParser.parse_model(
+            os.path.join(os.path.realpath(os.path.join(os.path.dirname(__file__), 'resources')),
+                         'CompoundMultDivWithDifferentButCompatibleUnits.nestml'))
+        printed_rhs_expression = print_rhs_of_first_assignment_in_update_block(model)
+        self.assertEqual(printed_rhs_expression, '1200*mV')
```
Comment (pnbabu) [inline tests/unit_system_test.py:119]: Isn't it better to add a coco test that errors out with such compound expressions?

Comment (clinssen) [reply tests/unit_system_test.py:119]: This is already covered by the type checking coco fixed by this PR. This tests checks that the coco picks up on it. I added a negative in addition to the positive test case to improve test coverage.

File: tests/valid/CompoundMultDivWithDifferentButCompatibleUnits.nestml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+You should have received a copy of the GNU General Public License
+along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+"""
+neuron MultDivTest:
+    state:
+        lhs V = 1 V
+    end
+
+    update:
+        lhs *= 1200mV
```
Comment (pnbabu) [inline tests/valid/CompoundMultDivWithDifferentButCompatibleUnits.nestml:32]: Is this a valid use case? The lhs is in `V` and the rhs is in `V*mV`

Comment (clinssen) [reply tests/valid/CompoundMultDivWithDifferentButCompatibleUnits.nestml]: This turned out to be an issue with test discovery, in particular, these compound operator .nestml files were not being processed by ``cocos_test.py``. This was fixed by 747bae7.

**Review by pnbabu — APPROVED** [review]
Looks good to me!

**Pull request merged**